src/rest/getStatusHandler.go

```diff
@@ -0,0 +1,77 @@
+package rest
+
+import (
+	"devicemanager/config"
+	"github.com/kataras/iris/v12"
+	"github.com/kataras/iris/v12/context"
+	"time"
+)
+
+const deviceManagerName = "DeviceManager"
+
+// StatusResponse holds the information of response of Device Manager Status
+type StatusResponse struct {
+	Comment         string          `json:"_comment"`
+	Name            string          `json:"Name"`
+	Version         string          `json:"Version"`
+	Status          Status          `json:"Status"`
+	EventMessageBus EventMessageBus `json:"EventMessageBus"`
+}
+
+// Status holds information of Device Manager Status
+type Status struct {
+	Available string `json:"Available"`
+	Uptime    string `json:"Uptime"`
+	TimeStamp string `json:"TimeStamp"`
+}
+
+// EventMessageBus holds the  information of  EMB Broker type and EMBQueue information
+type EventMessageBus struct {
+	EmbType  string     `json:"EmbType"`
+	EmbQueue []EmbQueue `json:"EmbQueue"`
+}
+
+// EmbQueue holds the  information of Queue Name and Queue Description
+type EmbQueue struct {
+	QueueName string `json:"EmbQueueName"`
+	QueueDesc string `json:"EmbQueueDesc"`
+}
+
+type statusHandler struct {
+	status          *Status
+	startupTime     time.Time
+	firmwareVersion string
+}
+
+func (s *Status) Init() {
+	s.RefreshTimeStampAndUptime(time.Now())
+	s.Available = "Yes"
+	s.Uptime = s.TimeStamp
```

Comment:
Why we assigning Uptime value twice? (l47 and l 54)

Reply:
Please note that first assignment is made during `Init()` method that is called only once.  Uptime is a value that indicates time passed since DM startup, so it needs to be constantly updated - this is the responsibility of `RefreshTimeStampAndUptime(startupTime time.Time)` method.

Reply:
ok, thanks for clarifying.